grpc/adapters.go

```diff
@@ -0,0 +1,44 @@
+package grpc
+
+import (
+	"fmt"
+
+	"github.com/ledgerhq/lama-bitcoin-svc/pb/v1"
+	"github.com/ledgerhq/lama-bitcoin-svc/pkg/bitcoin"
+)
+
+// ErrUnknownNetwork is returned when a string representing an unknown network
+// is found.
+type ErrUnknownNetwork string
+
+func (e ErrUnknownNetwork) Error() string {
+	return fmt.Sprintf("invalid network in chain params: %s", string(e))
+}
+
+func getBitcoinChainParams(chainParams *pb.ChainParams) (bitcoin.ChainParams, error) {
```

Comment:
As it is a wrapping factory, you can drop the `get` prefix, and simply write `func BitCoinChainParams...` or `func NewBitCoinChainParam...` if you want to stick with the factory method pattern.
It's not recommanded to have private factories to public types.